### PR TITLE
Lps 153483

### PR DESCRIPTION
**Pull request by shuyangzhou**
@david-truong to turn this on, set "virtual.hosts.strict.access=true" in portal-ext.properties
Then you can only access the portal via existing virtual hosts, except hosts listed in "virtual.hosts.ignore.hosts" portal property which is an existing one not added in this pull. Its default value is ```    virtual.hosts.ignore.hosts=\
        127.0.0.1,\
        localhost```
This means if someone messes up with virtual.hosts.ignore.hosts initially, then portal won't be accessible anymore. Use this with caution.

### CODE REVIEW

**Comment by shuyangzhou** [comment]
ci:reopen

File: portal-web/docroot/errors/code.jsp [hunk 1]
```diff
@@ -43,6 +43,9 @@ String xRequestWith = request.getHeader(HttpHeaders.X_REQUESTED_WITH);
 %>
 
 <c:choose>
+	<c:when test="<%= GetterUtil.getBoolean(request.getAttribute(WebKeys.UNKNOWN_VIRTUAL_HOST)) %>">
+		Unknown virtual host name : <%= PortalUtil.getHost(request) %>
```
Comment (shuyangzhou) [inline portal-web/docroot/errors/code.jsp:47]: @david-truong if you want some one from front end to make the 404 page prettier, do it here.

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/b4da5670e7...5788e2469b

**Comment by brianchandotcom** [comment]
@shuyangzhou see my SF/renames in my origin.

**Pull request closed without merging**